guides/hack/03-expressions-and-operators/97-assignment.md

```diff
@@ -62,4 +62,31 @@ $list[$i++] += 10;
 
 When the left-hand expression contains a side-effect, that side-effect is performed only once.
 
-The complete set of compound-assignment operators is: `**=`, `*=`, `/=`, `%=`, `+=`, `-=`, `.=`, `<<;=`, `>>=`, `&=`, `^=`, and `|=`.
+The complete set of compound-assignment operators is: `??=`, `**=`, `*=`, `/=`, `%=`, `+=`, `-=`, `.=`, `<<;=`, `>>=`, `&=`, `^=`, and `|=`.
+
+## Operator ??=
+
+The `??=` operator can be used for conditionally writing to a variable if it is null or not _set_.
+The rules around when something is _set_ are rather complex.
```

Comment:
Could we state the rules rather than saying they're complex and giving some examples?

I believe that for a variable, it's just whether it's `null`. For a Hack array or PHP array, it's whether the value is present and `null`, or not present.

If you split up your example and add a sentence, that would be perfect :)

Reply:
I have expanded `Operator ??`. That is a better place for these definitions.